File: src/main/java/me/bristermitten/plumber/scheduling/TaskFactory.java
package me.bristermitten.plumber.scheduling;

import com.google.inject.assistedinject.Assisted;
import me.bristermitten.plumber.scheduling.timings.Time;

/**
 * Guice Assisted Injection factory for creating instances of {@link Task}
 */
public interface TaskFactory {
    Task create(@Assisted("delay") long delay, @Assisted("period") long period, Runnable run);

    Task create(@Assisted("delay") Time delay, @Assisted("period") Time period, Runnable run);
}

File: src/main/kotlin/me/bristermitten/plumber/scheduling/TaskFactory.kt
package me.bristermitten.plumber.scheduling

import com.google.inject.assistedinject.Assisted
import me.bristermitten.plumber.scheduling.timings.Time

/**
 * Guice Assisted Injection factory for creating instances of [Task]
 */
interface TaskFactory {
    fun create(@Assisted("delay") delay: Long, @Assisted("period") period: Long, run: Runnable): Task

    fun create(@Assisted("delay") delay: Time, @Assisted("period") period: Time, run: Runnable): Task
}
